pandas/core/resample.py

```diff
@@ -852,7 +854,7 @@ def __init__(self, obj, *args, **kwargs):
         self._groupby.grouper.mutated = True
         self.groupby = copy.copy(parent.groupby)
 
-    def _apply(self, f, **kwargs):
+    def _apply(self, f, *args, **kwargs):
```

Comment:
`*args` is unused now, which seems incorrect. Shouldn't it be passed through?

Reply:
@TomAugspurger 
I added it for a compatibility and `TestResamplerGrouper.test_apply` failed

Reply:
Can you add a test that the proper behavior is happening now?

Reply:
@TomAugspurger 

Could you please clarify which test did you mean?

Anyway, at this moment, `_GroupByMixin` and `Resampler` have different arguments in `_groupby_and_aggregate`, so I added `args` for a compatibility.

Reply:
This seems to be hit by `df.groupby(...).resample`.

```python
In [1]: import pandas as pd

In [2]: def f(x, arg):
   ...:     assert arg == 0
   ...:     return arg
   ...:
   ...:

In [3]: df = pd.DataFrame({"A": 1, "B": 2}, index=pd.date_range('2017', periods=10))

In [4]: df.groupby("A").resample("D").agg(f, 0)
```

```pytb
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/sandbox/pandas/pandas/core/groupby/generic.py in aggregate(self, func_or_funcs, *args, **kwargs)
    774             try:
--> 775                 return self._python_agg_general(func_or_funcs, *args, **kwargs)
    776             except Exception:

~/sandbox/pandas/pandas/core/groupby/groupby.py in _python_agg_general(self, func, *args, **kwargs)
    839         if len(output) == 0:
--> 840             return self._python_apply_general(f)
    841

~/sandbox/pandas/pandas/core/groupby/groupby.py in _python_apply_general(self, f)
    701         keys, values, mutated = self.grouper.apply(f, self._selected_obj,
--> 702                                                    self.axis)
    703

~/sandbox/pandas/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    197             group_axes = _get_axes(group)
--> 198             res = f(group)
    199             if not _is_indexed_like(res, group_axes):

~/sandbox/pandas/pandas/core/groupby/groupby.py in <lambda>(x)
    827         func = self._is_builtin_func(func)
--> 828         f = lambda x: func(x, *args, **kwargs)
    829

TypeError: f() missing 1 required positional argument: 'arg'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
~/sandbox/pandas/pandas/core/resample.py in _groupby_and_aggregate(self, how, grouper, *args, **kwargs)
    325                 # Check if the function is reducing or not.
--> 326                 result = grouped._aggregate_item_by_item(how, *args, **kwargs)
    327             else:

~/sandbox/pandas/pandas/core/groupby/generic.py in _aggregate_item_by_item(self, func, *args, **kwargs)
    279             if not len(result_columns) and errors is not None:
--> 280                 raise errors
    281

~/sandbox/pandas/pandas/core/groupby/generic.py in _aggregate_item_by_item(self, func, *args, **kwargs)
    264                 result[item] = self._try_cast(
--> 265                     colg.aggregate(func, *args, **kwargs), data)
    266             except ValueError:

~/sandbox/pandas/pandas/core/groupby/generic.py in aggregate(self, func_or_funcs, *args, **kwargs)
    776             except Exception:
--> 777                 result = self._aggregate_named(func_or_funcs, *args, **kwargs)
    778

~/sandbox/pandas/pandas/core/groupby/generic.py in _aggregate_named(self, func, *args, **kwargs)
    904             group.name = name
--> 905             output = func(group, *args, **kwargs)
    906             if isinstance(output, (Series, Index, np.ndarray)):

TypeError: f() missing 1 required positional argument: 'arg'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
~/sandbox/pandas/pandas/core/groupby/groupby.py in apply(self, func, *args, **kwargs)
    683             try:
--> 684                 result = self._python_apply_general(f)
    685             except Exception:

~/sandbox/pandas/pandas/core/groupby/groupby.py in _python_apply_general(self, f)
    701         keys, values, mutated = self.grouper.apply(f, self._selected_obj,
--> 702                                                    self.axis)
    703

~/sandbox/pandas/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    197             group_axes = _get_axes(group)
--> 198             res = f(group)
    199             if not _is_indexed_like(res, group_axes):

TypeError: f() missing 1 required positional argument: 'arg'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
~/sandbox/pandas/pandas/core/groupby/groupby.py in apply(self, func, *args, **kwargs)
    683             try:
--> 684                 result = self._python_apply_general(f)
    685             except Exception:

~/sandbox/pandas/pandas/core/groupby/groupby.py in _python_apply_general(self, f)
    701         keys, values, mutated = self.grouper.apply(f, self._selected_obj,
--> 702                                                    self.axis)
    703

~/sandbox/pandas/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    197             group_axes = _get_axes(group)
--> 198             res = f(group)
    199             if not _is_indexed_like(res, group_axes):

~/sandbox/pandas/pandas/core/resample.py in func(x)
    869
--> 870             return x.apply(f, **kwargs)
    871

~/sandbox/pandas/pandas/core/resample.py in aggregate(self, func, *args, **kwargs)
    246                                                  *args,
--> 247                                                  **kwargs)
    248

~/sandbox/pandas/pandas/core/resample.py in _groupby_and_aggregate(self, how, grouper, *args, **kwargs)
    332             # try to evaluate
--> 333             result = grouped.apply(how, *args, **kwargs)
    334

~/sandbox/pandas/pandas/core/groupby/groupby.py in apply(self, func, *args, **kwargs)
    695                 with _group_selection_context(self):
--> 696                     return self._python_apply_general(f)
    697

~/sandbox/pandas/pandas/core/groupby/groupby.py in _python_apply_general(self, f)
    701         keys, values, mutated = self.grouper.apply(f, self._selected_obj,
--> 702                                                    self.axis)
    703

~/sandbox/pandas/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    197             group_axes = _get_axes(group)
--> 198             res = f(group)
    199             if not _is_indexed_like(res, group_axes):

TypeError: f() missing 1 required positional argument: 'arg'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
~/sandbox/pandas/pandas/core/resample.py in _groupby_and_aggregate(self, how, grouper, *args, **kwargs)
    325                 # Check if the function is reducing or not.
--> 326                 result = grouped._aggregate_item_by_item(how, *args, **kwargs)
    327             else:

~/sandbox/pandas/pandas/core/groupby/generic.py in _aggregate_item_by_item(self, func, *args, **kwargs)
    279             if not len(result_columns) and errors is not None:
--> 280                 raise errors
    281

~/sandbox/pandas/pandas/core/groupby/generic.py in _aggregate_item_by_item(self, func, *args, **kwargs)
    264                 result[item] = self._try_cast(
--> 265                     colg.aggregate(func, *args, **kwargs), data)
    266             except ValueError:

~/sandbox/pandas/pandas/core/groupby/generic.py in aggregate(self, func_or_funcs, *args, **kwargs)
    776             except Exception:
--> 777                 result = self._aggregate_named(func_or_funcs, *args, **kwargs)
    778

~/sandbox/pandas/pandas/core/groupby/generic.py in _aggregate_named(self, func, *args, **kwargs)
    904             group.name = name
--> 905             output = func(group, *args, **kwargs)
    906             if isinstance(output, (Series, Index, np.ndarray)):

TypeError: f() missing 1 required positional argument: 'arg'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
~/sandbox/pandas/pandas/core/groupby/groupby.py in apply(self, func, *args, **kwargs)
    683             try:
--> 684                 result = self._python_apply_general(f)
    685             except Exception:

~/sandbox/pandas/pandas/core/groupby/groupby.py in _python_apply_general(self, f)
    701         keys, values, mutated = self.grouper.apply(f, self._selected_obj,
--> 702                                                    self.axis)
    703

~/sandbox/pandas/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    197             group_axes = _get_axes(group)
--> 198             res = f(group)
    199             if not _is_indexed_like(res, group_axes):

TypeError: f() missing 1 required positional argument: 'arg'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-4-9cc8965c7cbe> in <module>()
----> 1 df.groupby("A").resample("D").agg(f, 0)

~/sandbox/pandas/pandas/core/resample.py in aggregate(self, func, *args, **kwargs)
    245                                                  grouper,
    246                                                  *args,
--> 247                                                  **kwargs)
    248
    249         result = self._apply_loffset(result)

~/sandbox/pandas/pandas/core/resample.py in _apply(self, f, *args, **kwargs)
    870             return x.apply(f, **kwargs)
    871
--> 872         result = self._groupby.apply(func)
    873         return self._wrap_result(result)
    874

~/sandbox/pandas/pandas/core/groupby/groupby.py in apply(self, func, *args, **kwargs)
    694
    695                 with _group_selection_context(self):
--> 696                     return self._python_apply_general(f)
    697
    698         return result

~/sandbox/pandas/pandas/core/groupby/groupby.py in _python_apply_general(self, f)
    700     def _python_apply_general(self, f):
    701         keys, values, mutated = self.grouper.apply(f, self._selected_obj,
--> 702                                                    self.axis)
    703
    704         return self._wrap_applied_output(

~/sandbox/pandas/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    196             # group might be modified
    197             group_axes = _get_axes(group)
--> 198             res = f(group)
    199             if not _is_indexed_like(res, group_axes):
    200                 mutated = True

~/sandbox/pandas/pandas/core/resample.py in func(x)
    868                 return getattr(x, f)(**kwargs)
    869
--> 870             return x.apply(f, **kwargs)
    871
    872         result = self._groupby.apply(func)

~/sandbox/pandas/pandas/core/resample.py in aggregate(self, func, *args, **kwargs)
    245                                                  grouper,
    246                                                  *args,
--> 247                                                  **kwargs)
    248
    249         result = self._apply_loffset(result)

~/sandbox/pandas/pandas/core/resample.py in _groupby_and_aggregate(self, how, grouper, *args, **kwargs)
    331             # we have a non-reducing function
    332             # try to evaluate
--> 333             result = grouped.apply(how, *args, **kwargs)
    334
    335         result = self._apply_loffset(result)

~/sandbox/pandas/pandas/core/groupby/groupby.py in apply(self, func, *args, **kwargs)
    694
    695                 with _group_selection_context(self):
--> 696                     return self._python_apply_general(f)
    697
    698         return result

~/sandbox/pandas/pandas/core/groupby/groupby.py in _python_apply_general(self, f)
    700     def _python_apply_general(self, f):
    701         keys, values, mutated = self.grouper.apply(f, self._selected_obj,
--> 702                                                    self.axis)
    703
    704         return self._wrap_applied_output(

~/sandbox/pandas/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    196             # group might be modified
    197             group_axes = _get_axes(group)
--> 198             res = f(group)
    199             if not _is_indexed_like(res, group_axes):
    200                 mutated = True

TypeError: f() missing 1 required positional argument: 'arg'

```

So IIUC, we want to be passing through `*args` here.